### PR TITLE
Add storageclass check

### LINKED ISSUE #15: the strategy about setting mongodb storageclass
In document we need to ask users to set the default storageclass before installation and tell them if they don't set one we will automatically choose one:

https://docs.openshift.com/container-platform/4.1/storage/dynamic-provisioning.html#defining-storage-classes_dynamic-provisioning

The logic to choose storage class can be:

1. If the cluster has no storage class, log error message and stuck in reconcile loop
1. If there is only one default storageclass, then we don't set storageclass and let the kubernetes assign one to mongodb
1. If there are multiple default storage classes, then we choose the first one and set to sts
1. If there is no default storage class, we choose the first one and set it to mongodb sts

### PR DESCRIPTION
**Pull request by horis233**
fix #15

### CODE REVIEW

File: pkg/controller/mongodb/mongodb_controller.go [hunk 1]
```diff
@@ -298,3 +312,29 @@ func (r *ReconcileMongoDB) createFromYaml(instance *operatorv1alpha1.MongoDB, ya
...
 }
+
+func (r *ReconcileMongoDB) getstorageclass() (string, error) {
+	scList := &storagev1.StorageClassList{}
+	err := r.client.List(context.TODO(), scList)
+	if err != nil {
+		return "", err
+	}
+	if len(scList.Items) == 0 {
+		return "", fmt.Errorf("could not storage class in the cluster")
```
Comment (chenzhiwei) [inline pkg/controller/mongodb/mongodb_controller.go]: `could not find`

File: pkg/controller/mongodb/mongodb_controller.go [hunk 2]
```diff
@@ -298,3 +312,29 @@ func (r *ReconcileMongoDB) createFromYaml(instance *operatorv1alpha1.MongoDB, ya
...
+	}
+	for _, sc := range scList.Items {
```
Comment (chenzhiwei) [inline pkg/controller/mongodb/mongodb_controller.go]: can we use a single loop to get all the sc and then choose one? We can later reuse this code if there is strategy change.

```
    var defaultSC []string
    var nonDefaultSC []string
    for _, sc := range scList.Items {
        if sc.Provisioner == "kubernetes.io/no-provisioner" {
            continue
        }
        if sc.ObjectMeta.GetAnnotations()["storageclass.beta.kubernetes.io/is-default-class"] == "true" {
            defaultSC = append(defaultSC, sc.GetName())
            continue
        }
        nonDefaultSC = append(nonDefaultSC, sc.GetName())
    }

    if len(defaultSC) != 0 {
        return defaultSC[1], nil
    }

    if len(nonDefaultSC) != 0 {
        return nonDefaultSC[1], nil
    }
```

**Comment by horis233** [comment]
> please check inline comments

Thanks, updated.

**Comment by chenzhiwei** [comment]
Sorry for misleading this PR.

The storageclass is cluster scoped resource, the mongodb-operator has no permission to access it.

I will create another PR to use the default storageClass or explicitly set one.

**Pull request merged**